Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1278,22 +1278,16 @@ StyleDifference RenderStyle::diff(const RenderStyle& other, OptionSet<StyleDiffe
 {
     changedContextSensitiveProperties = OptionSet<StyleDifferenceContextSensitiveProperty>();
 
-    StyleDifference svgChange = StyleDifference::Equal;
-    if (m_svgStyle != other.m_svgStyle) {
-        svgChange = m_svgStyle->diff(other.m_svgStyle.get());
-        if (svgChange == StyleDifference::Layout)
-            return svgChange;
-    }
+    bool currentColorDiffers = m_inheritedData->color != other.m_inheritedData->color;
```

Comment:
I think splitting the function in the same way in RenderStyle/SVGRenderStyle provides more flexibility (for example now, the repaint change from SVG only occurs after the RepaintLayer or the LayoutPositionedMovementOnly). This would be cumbersome to write with only the diff function.